dial.go

```diff
@@ -390,15 +528,15 @@ func readGreeting(reader io.Reader) (string, string, error) {
 
 // identify sends info about client protocol, receives info
 // about server protocol in response and stores it in the connection.
-func identify(w writeFlusher, r io.Reader) (ProtocolInfo, error) {
+func identify(conn protocolConn) (ProtocolInfo, error) {
```

Comment:
It should work I think.
```suggestion
func identify(conn Conn) (ProtocolInfo, error) {
```